test/test_cover.sh

```diff
@@ -3,11 +3,34 @@
 PKGS=$(go list github.com/tendermint/tendermint/... | grep -v /vendor/)
 
 set -e
+
 echo "mode: atomic" > coverage.txt
-for pkg in ${PKGS[@]}; do
-	go test -v -timeout 30m -race -coverprofile=profile.out -covermode=atomic "$pkg"
-	if [ -f profile.out ]; then
-		tail -n +2 profile.out >> coverage.txt;
-		rm profile.out
-	fi
-done
+if [ $1 == "fast" ]; then 
+	for pkg in ${PKGS[@]}; do
+		echo $pkg
+		if [ $pkg != "github.com/tendermint/tendermint/consensus" ]; then
+			if [ $pkg != "github.com/tendermint/tendermint/node" ]; then
+				if [ $pkg != "github.com/tendermint/tendermint/rpc/client" ]; then
+					go test -v -timeout 30m -race -coverprofile=profile.out -covermode=atomic "$pkg"
+					if [ -f profile.out ]; then
+						tail -n +2 profile.out >> coverage.txt;
+						rm profile.out
+					fi
+				fi
+			fi
+		fi
+	done
+fi
+
+if [ $1 == "slow" ]; then 
+	for pkg in ${PKGS[@]}; do
+		echo $pkg
```

Comment:
not sure what determiend these distinctions. `types` runs way slower on my machine than `node` or `rpc/client`. 

Is this really helpful ? Why not just run them all together as `make test` ...

Reply:
i split them up because of non-deterministic failures in the "slow" tests. Sure, another package might be slower or faster; these three "slow" packages just really mean "takes relatively long to run and has been known to reliably fail non-deterministically

Reply:
it's much easier to debug failures when they're in different jobs. The "slow" tests also generally pump out tons of logs. IMO the goal should be to, yeah, move the slow tests into the fast section, once they're sped up and more deterministic.

Reply:
great thanks for explanation!